model_analyzer/config/run/model_run_config.py

```diff
@@ -115,6 +115,37 @@ def is_legal_combination(self):
             logger.debug(
                 f"Illegal model run config because client batch size {perf_batch_size} is greater than model max batch size {max_batch_size}"
             )
+
+        return legal
+
+    def _check_for_preferred_vs_model_batch_size(self):
```

Comment:
please add unit tests (you could do it in test_run_config.py, and give it one model_config)

Reply:
Added unit tests for mrc.is_legal_combination()

Reply:
Tests look good!